[SRVCOM-2667] Improve the Serverless Overview page

ochromy opened the pull request:
Version(s):
serverless-docs-1.33+

Issue:
https://issues.redhat.com/browse/SRVCOM-2667

Link to docs preview:
- https://81607--ocpdocs-pr.netlify.app/openshift-serverless/latest/about/about-serverless.html

QE review:
- [x] QE has approved this change.

Additional information:
Updated the Overview page with the descriptions from the operator Details + Logic

mgencur commented:
LGTM from QE

ochromy commented:
/label peer-review-needed
/label serverless

lpettyjo reviewed (commented):
LGTM!

ochromy commented:
Thank you both! 
/label merge-review-needed

ochromy commented:
/label merge-review-needed

michaelryanpeter commented:
/label merge-review-in-progress

michaelryanpeter commented on about/about-serverless.adoc:
```diff
@@ -25,11 +24,49 @@ For additional information about the {ServerlessProductName} life cycle and supp
+{ServerlessLogicProductName} implements the link:https://serverlessworkflow.io/[CNCF Serverless Workflow specification].
+
+=== Knative CLI
+The Knative client `kn` allows you to create Knative resources from the command line or from within Shell scripts. With its extensive help pages and autocompletion support, it frees you from memorizing the detailed structure of the Knative resource schemas. `kn` features include:
```
```suggestion
The Knative (`kn`) CLI allows you to create Knative resources from the command line or from within Shell scripts.  The Knative (`kn`) CLI includes the following features:
```
Minimalism suggestion for fluff. Formatting suggestions based on the [Knative CLI docs](https://docs.openshift.com/serverless/1.33/cli_tools/kn-plugins.html).

michaelryanpeter commented on about/about-serverless.adoc:
```diff
@@ -25,11 +24,49 @@ For additional information about the {ServerlessProductName} life cycle and supp
+=== Knative CLI
+The Knative client `kn` allows you to create Knative resources from the command line or from within Shell scripts. With its extensive help pages and autocompletion support, it frees you from memorizing the detailed structure of the Knative resource schemas. `kn` features include:
+
+* Full support for managing all features of Knative Serving: Services, Revisions and Routes
+* Support for managing Knative Eventing entities: Sources, Brokers, Triggers, Channels and Subscriptions
```
This would scan better and have a more parallel structure if you used subbullets to create a nested list.

See [Nested unordered lists](https://docs.asciidoctor.org/asciidoc/latest/lists/unordered/#nested-unordered-list) (Asciidoc docs)

michaelryanpeter commented on about/about-serverless.adoc:
```diff
@@ -25,11 +24,49 @@ For additional information about the {ServerlessProductName} life cycle and supp
+
+* Full support for managing all features of Knative Serving: Services, Revisions and Routes
+* Support for managing Knative Eventing entities: Sources, Brokers, Triggers, Channels and Subscriptions
+* A kubectl-like plugin architecture to extend the built-in functionality
```
```suggestion
* A plugin architecture based on the design of the Kubernetes (`kubectl`) CLI tool
```

michaelryanpeter commented on about/about-serverless.adoc:
```diff
@@ -25,11 +24,49 @@ For additional information about the {ServerlessProductName} life cycle and supp
+=== Knative CLI
+The Knative client `kn` allows you to create Knative resources from the command line or from within Shell scripts. With its extensive help pages and autocompletion support, it frees you from memorizing the detailed structure of the Knative resource schemas. `kn` features include:
+
+* Full support for managing all features of Knative Serving: Services, Revisions and Routes
```
```suggestion
* Support for managing the following Knative Serving features: 
** Services
** Revisions
** Routes
```

michaelryanpeter commented on about/about-serverless.adoc:
```diff
@@ -25,11 +24,49 @@ For additional information about the {ServerlessProductName} life cycle and supp
+* Full support for managing all features of Knative Serving: Services, Revisions and Routes
+* Support for managing Knative Eventing entities: Sources, Brokers, Triggers, Channels and Subscriptions
+* A kubectl-like plugin architecture to extend the built-in functionality
+* Easy integration of Knative into Tekton pipelines by using kn in a Tekton task
```
```suggestion
* Integration of Knative into Tekton pipelines
```
See the ISG a[bout the term _easy_](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#easy). 
Minimalism: the last clause isn't adding much. If the part about Tetkon tasks is more important, consider leading with that part.

michaelryanpeter reviewed (changes requested):
I left comments around some style, minimalism, grammar, and AsciiDoctor issues. 

If you are under a very tight deadline, please address the following issue before resubmitting your PR to the merge review queue:
* You must address the issue with headline ids for all the places where it is missing. 

There are some suggestions that are up to you if you want to apply them, but the places where I have cited the ISG, SSG, or repo guidelines need to be addressed. 

If you are under the pressure of a deadline, you can create a follow up ticket to track fixing the rest of the issues after your release.

Reach out if you have questions. For timely communication, it is best to reach me on Slack.

/remove-label peer-review-in-progress

michaelryanpeter commented:
Apologies, I did not mean to request changes.

michaelryanpeter commented:
Sorry, I had to approve the PR to undo the hold created by accidentally hitting the `Changes required` comment. We don't use that in the repo, and I apologize for my mistake. 

To summarize my review comment, the only merge blocker here is fixing the heading ids. It would be great if you can address the other comments now, but if you are under a tight deadline, feel free to create a follow up Jira to address the work at a later time. Thanks!

michaelryanpeter commented:
/remove-label merge-review-in-progress

michaelryanpeter commented:
/remove-label merge-review-needed

ochromy commented:
Hello @michaelryanpeter,
Thank you very much for the suggestions, references and catching the missing IDs. Apologies for that.
I've pushed the updated version. Please let me know if I addressed your comments correctly.

michaelryanpeter commented on about/about-serverless.adoc:
```diff
@@ -24,12 +23,77 @@ For additional information about the {ServerlessProductName} life cycle and supp
+
+Knative Serving builds on Kubernetes to support deploying and serving of applications and functions as serverless containers. Serving simplifies the application deployment, dynamically scales based on in incoming traffic and supports custom rollout strategies with traffic splitting. 
+
+Other features of Knative Serving include:
```
The important part from the IBM Style guide on lead-in sentences is to make sure the lead-in is a complete sentence.
```suggestion
Knative Serving includes the following features:
```

michaelryanpeter commented on about/about-serverless.adoc:
```diff
@@ -24,12 +23,77 @@ For additional information about the {ServerlessProductName} life cycle and supp
 
+{FunctionsProductName} allows you to write functions that are deployed as Knative Services and take advantage of Knative Serving and Eventing. 
+
+Other features of {FunctionsProductName} include:
```
See the comment above about using complete sentences to introduce a list.

michaelryanpeter reviewed (commented):
Two small nits, but you can follow up on those comments at another time. Are you ready for a merge @ochromy?

ochromy commented:
> Two small nits, but you can follow up on those comments at another time. Are you ready for a merge @ochromy?

@michaelryanpeter Yes, the commit can now be merged. And Thanks for the example of the lead-in sentence.

michaelryanpeter commented:
/cherrypick serverless-docs-1.33
